sms_no_automatic_delete/models/sms_sms.py

```diff
@@ -1,37 +1,42 @@
 # Copyright 2021 Akretion (https://www.akretion.com).
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 
+import logging
 from datetime import datetime, timedelta
 
-from odoo import api, models
+from odoo import models
+from odoo.tools.sql import SQL
+
+_logger = logging.getLogger(__name__)
 
 
 class SmsSms(models.Model):
     _inherit = "sms.sms"
 
-    @api.model
-    def _postprocess_iap_sent_sms(
-        self, iap_results, failure_reason=None, unlink_failed=False, unlink_sent=True
-    ):
-        # Update state of the sms as "sent" when they are sent.
-        ids = [item["res_id"] for item in iap_results if item["state"] == "success"]
-        self.browse(ids).write({"state": "sent"})
-        return super()._postprocess_iap_sent_sms(
-            iap_results,
-            failure_reason=failure_reason,
-            unlink_failed=unlink_failed,
-            unlink_sent=unlink_sent,
+    IAP_TO_SMS_STATE_SUCCESS = {
+        "processing": "process",
+        "success": "sent",
+        "sent": "sent",
+        "delivered": "sent",
+    }
+
+    def _gc_device(self):
```

Comment:
- Add @api.autovacuum : without it, the autovacuum could not be triggered 
- Seems that you did not test your work from UI.